lib/chef/sugar/rackspace/rackconnect.rb

```diff
@@ -6,16 +6,30 @@ module Rackconnect
       extend self
 
       def rackconnect_v2?(node)
-        return false unless node.key? 'rackspace'
-        if node['rackspace'].key? 'rackconnect'
-          return node['rackspace']['rackconnect']['enabled']
+        return false unless rackconnected?(node)
+        if node['rackspace']['rackconnect']['version'] == 2
+          return true
+        else
+          return false
+        end
+      end
+
+      def rackconnect_v3?(node)
+        return false unless rackconnected?(node)
+        if node['rackspace']['rackconnect']['version'] == 3
+          return true
         else
           return false
         end
       end
 
       def rackconnected?(node)
-        rackconnect_v2?(node)
+        return false unless node.key? 'rackspace'
+        if node['rackspace'].key? 'rackconnect'
```

Comment:
Could this just be `return node['rackspace'].key? 'rackconnect' && node['rackspace']['rackconnect']['enabled']`?